magicctapipe/scripts/lst1_magic/semi_automatic_scripts/stereo_events.py

```diff
@@ -0,0 +1,364 @@
+            f"#SBATCH --array=0-{process_size}%100\n",
+            "#SBATCH --mem=30g\n",
+            "#SBATCH -N 1\n\n",
+            "ulimit -l unlimited\n",
+            "ulimit -s unlimited\n",
+            "ulimit -a\n\n",
+            f"export INPUTDIR={inputdir}\n",
+            f"export OUTPUTDIR={inputdir}/StereoMerged\n",
+            "SAMPLE_LIST=($(<$INPUTDIR/list_coin.txt))\n",
+            "SAMPLE=${SAMPLE_LIST[${SLURM_ARRAY_TASK_ID}]}\n",
+            "export LOG=$OUTPUTDIR/stereo_${SLURM_ARRAY_TASK_ID}.log\n",
+            f"conda run -n {env_name} lst1_magic_stereo_reco --input-file $SAMPLE --output-dir $OUTPUTDIR --config-file {target_dir}/config_stereo.yaml >$LOG 2>&1",
+        ]
+        f.writelines(lines)
+
+
+def main():
+
+    """
+    Here we read the config_general.yaml file and call the functions defined above.
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--config-file",
+        "-c",
+        dest="config_file",
+        type=str,
+        default="./config_general.yaml",
+        help="Path to a configuration file",
+    )
+
+    parser.add_argument(
+        "--analysis-type",
+        "-t",
+        choices=["onlyReal", "onlyMC"],
+        dest="analysis_type",
+        type=str,
+        default="doEverything",
+        help="You can type 'onlyReal' or 'onlyMC' to run this script only on real or MC data, respectively.",
+    )
+
+    args = parser.parse_args()
+    with open(
+        args.config_file, "rb"
+    ) as f:  # "rb" mode opens the file in binary format for reading
+        config = yaml.safe_load(f)
+
+    target_dir = f'{Path(config["directories"]["workspace_dir"])}/{config["directories"]["target_name"]}'
+
+    env_name = config["general"]["env_name"]
+    source = config["directories"]["target_name"]
+    NSB_match = config["general"]["NSB_matching"]
+    telescope_ids = list(config["mc_tel_ids"].values())
+
+    print("***** Generating file config_stereo.yaml...")
+    print("***** This file can be found in ", target_dir)
+    configfile_stereo(telescope_ids, target_dir)
+
+    # Below we run the analysis on the MC data
+    if (
+        (args.analysis_type == "onlyMC")
+        or (args.analysis_type == "doEverything")
+        and not NSB_match
+    ):
+        print("***** Generating the bashscript for MCs...")
+        bash_stereoMC(target_dir, "gammadiffuse", env_name)
+        bash_stereoMC(target_dir, "gammas", env_name)
+        bash_stereoMC(target_dir, "protons", env_name)
+        bash_stereoMC(target_dir, "protons_test", env_name)
+
+        list_of_stereo_scripts = np.sort(glob.glob("StereoEvents_MC_*.sh"))
+
+        for n, run in enumerate(list_of_stereo_scripts):
+            if n == 0:
+                launch_jobs = f"stereo{n}=$(sbatch --parsable {run})"
+            else:
+                launch_jobs = f"{launch_jobs} && stereo{n}=$(sbatch --parsable --dependency=afterany:$stereo{n-1} {run})"
```

Comment:
why do you have a dependency here on the previous job? shouldn't they be independent as they are simply different runs to be processed?